packages/neuron-ui/src/widgets/Calendar/index.tsx

```diff
@@ -0,0 +1,186 @@
+                  onClick={() => onChange(date.instance)}
+                >
+                  {date.date}
+                </button>
+              </td>
+            ))}
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  )
+
+  const monthOptions: Option[] = [...Array(12).keys()].map(index => ({
+    value: index + 1,
+    title: locale.monthNames[index],
+    selectable: monthInRange(year, index, { minDate, maxDate }),
+  }))
+  const yearOptions: Option[] = [...Array(12).keys()].map(index => ({
+    value: year - 6 + index,
+    title: `${year - 6 + index}`,
+    selectable: yearInRange(year - 6 + index, { minDate, maxDate }),
+  }))
+
+  const prevMonth = () => {
+    if (month > 1) {
+      setMonth(m => m - 1)
+    } else {
+      setYear(y => y - 1)
+      setMonth(12)
+    }
+  }
+  const nextMonth = () => {
+    if (month < 12) {
+      setMonth(m => m + 1)
+    } else {
+      setYear(y => y + 1)
+      setMonth(1)
+    }
+  }
+
+  const calendarHeader = (
+    <div className={styles.calendarHeader}>
+      <button type="button" aria-label="prev" title="prev" className={styles.calPrev} onClick={prevMonth}>
+        {'<'}
+      </button>
+      <div className={styles.calTitle}>
+        <button
+          type="button"
+          aria-label={monthname}
+          title={monthname}
+          aria-haspopup="true"
+          aria-controls="menu"
+          onClick={() => setStatus('month')}
+        >
+          {monthname}
+        </button>
+        <button
+          type="button"
+          aria-label={`${year}`}
+          title={`${year}`}
+          aria-haspopup="true"
+          aria-controls="menu"
+          onClick={() => setStatus('year')}
+        >
+          {year}
+        </button>
+      </div>
+      <button type="button" aria-label="next" title="next" className={styles.calNext} onClick={nextMonth}>
+        {'>'}
+      </button>
+    </div>
+  )
+
+  const onChangeMonth = (monthOptionItem: Option) => {
+    setMonth(monthOptionItem.value)
+    setStatus('date')
+  }
+  const onChangeYear = (yearOptionItem: Option) => {
+    setYear(yearOptionItem.value)
+    setStatus('month')
+  }
+
+  return (
+    <div className={styles.calendar}>
+      {calendarHeader}
+      {status === 'date' && calendarTable}
+      {status === 'year' && <Selector options={yearOptions} onChange={onChangeYear} />}
+      {status === 'month' && <Selector options={monthOptions} onChange={onChangeMonth} />}
+    </div>
+  )
+}
+
+export default React.memo(Calendar, (prevProps, nextProps) => {
+  if (prevProps.value?.toDateString() !== nextProps.value?.toDateString()) {
+    return false
+  }
+  if (prevProps.minDate?.toDateString() !== nextProps.minDate?.toDateString()) {
+    return false
+  }
+  if (prevProps.maxDate?.toDateString() !== nextProps.maxDate?.toDateString()) {
+    return false
+  }
+  if (prevProps.onChange !== nextProps.onChange) {
+    return false
+  }
+  return true
```

Comment:
```suggestion
  return prevProps.value?.toDateString() === nextProps.value?.toDateString()
  && prevProps.minDate?.toDateString() === nextProps.minDate?.toDateString()
  && prevProps.maxDate?.toDateString() === nextProps.maxDate?.toDateString()
  && prevProps.onChange === nextProps.onChange
```

On the other hand, I think we can let whoever uses this component keep the reference equal.